Realm/Realm/Realm.cs

```diff
@@ -595,7 +595,85 @@ public void Add<T>(IEnumerable<T> objs, bool update = false)
+        /// meaning that after it has run, <see cref="AsymmetricObject"/> will be managed.
+        /// Once an <see cref="AsymmetricObject"/> becomes managed dereferencing any property
+        /// of the original <see cref="AsymmetricObject"/> reference throws an exception.
+        /// </remarks>
+        public void Add<T>(T obj)
+            where T : IAsymmetricObject
+        {
+            ThrowIfDisposed();
+            Argument.NotNull(obj, nameof(obj));
+            Argument.Ensure(!obj.IsManaged, $"{nameof(obj)} must not be already managed by a Realm.", nameof(obj));
+
+            AddInternal(obj, obj.GetType(), update: false);
+        }
+
+        /// <summary>
+        /// Add a collection of standalone <see cref="AsymmetricObject"/>s to this <see cref="Realm"/>.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The Type T must not only be a <see cref="AsymmetricObject"/> but also have been processed by the Fody weaver,
+        /// so it has persistent properties.
+        /// </typeparam>
+        /// <param name="objs">A collection of <see cref="RealmObject"/> instances that will be added to this <see cref="Realm"/>.</param>
+        /// <exception cref="RealmInvalidTransactionException">
+        /// If you invoke this when there is no write <see cref="Transaction"/> active on the <see cref="Realm"/>.
+        /// </exception>
+        /// <exception cref="RealmObjectManagedByAnotherRealmException">
+        /// You can't manage an object with more than one <see cref="Realm"/>.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        /// If you invoke this method on a closed <see cref="Realm"/>.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// If you pass a null argument to this method.
+        /// </exception>
+        /// <remarks>
+        /// If the collection contains items that are already managed by this <see cref="Realm"/>, they will be ignored.
+        /// This method modifies the objects in-place, meaning that after it has run, all items in the collection will be managed.
+        /// Once an <see cref="AsymmetricObject"/> becomes managed and the transaction is committed,
+        /// dereferencing any property of the original <see cref="AsymmetricObject"/> reference throw an exeception.
+        /// Hence, none of the properties of the elements in the collection can be deferenced anymore after the transaction.
+        /// </remarks>
+        public void Add<T>(IEnumerable<T> objs)
+            where T : IAsymmetricObject
+        {
+            ThrowIfDisposed();
+            Argument.NotNull(objs, nameof(objs));
+            Argument.Ensure(objs.All(o => o != null), $"{nameof(objs)} must not contain null values.", nameof(objs));
+            Argument.Ensure(objs.All(o => !o.IsManaged), $"{nameof(objs)} must not contain already managed objects by a Realm.", nameof(objs));
```

Comment:
This is inefficient because we're iterating the collection twice. We could instead have a foreach loop that checks for both conditions.